src/gtkspellcheck/spellcheck.py

```diff
@@ -495,121 +477,181 @@ def check_range(self, start, end, force_all=False):
         return menu
 
     def _suggestion_menu(self, word):
         menu = []
         suggestions = self._dictionary.suggest(word)
         if not suggestions:
-            if _pygobject:
-                item = gtk.MenuItem.new()
-                label = gtk.Label.new("")
-            else:
-                item = gtk.MenuItem()
-                label = gtk.Label()
-            try:
-                label.set_halign(gtk.Align.LEFT)
-            except AttributeError:
-                label.set_alignment(0.0, 0.5)
-            label.set_markup("<i>{text}</i>".format(text=_("(no suggestions)")))
-            item.add(label)
-            menu.append(item)
-        else:
-            for suggestion in suggestions:
-                if _pygobject:
-                    item = gtk.MenuItem.new()
-                    label = gtk.Label.new("")
-                else:
-                    item = gtk.MenuItem()
-                    label = gtk.Label()
-                label.set_markup("<b>{text}</b>".format(text=suggestion))
+            if Gtk.MAJOR_VERSION < 4:
+                item = Gtk.MenuItem.new()
+                label = Gtk.Label.new("")
                 try:
-                    label.set_halign(gtk.Align.LEFT)
+                    label.set_halign(Gtk.Align.LEFT)
                 except AttributeError:
                     label.set_alignment(0.0, 0.5)
+                label.set_markup("<i>{text}</i>".format(text=_("(no suggestions)")))
                 item.add(label)
-                item.connect("activate", self._replace_word, word, suggestion)
                 menu.append(item)
-        if _pygobject:
-            menu.append(gtk.SeparatorMenuItem.new())
-            item = gtk.MenuItem.new_with_label(_('Add "{}" to Dictionary').format(word))
         else:
-            menu.append(gtk.SeparatorMenuItem())
-            item = gtk.MenuItem(_('Add "{}" to Dictionary').format(word))
-        item.connect("activate", lambda *args: self.add_to_dictionary(word))
+            for suggestion in suggestions:
+                if Gtk.MAJOR_VERSION > 3:
+                    escaped = suggestion.replace("'", "\\'")
+                    item = Gio.MenuItem.new(
+                        suggestion, f"spelling.replace-word('{escaped}')"
+                    )
+                else:
+                    item = Gtk.MenuItem.new()
+                    label = Gtk.Label.new("")
+                    label.set_markup("<b>{text}</b>".format(text=suggestion))
+                    try:
+                        label.set_halign(Gtk.Align.LEFT)
+                    except AttributeError:
+                        label.set_alignment(0.0, 0.5)
+                    item.add(label)
+                    item.connect(
+                        "activate", lambda *args: self._replace_word(suggestion)
+                    )
+                menu.append(item)
+        add_to_dict_menu_label = _('Add "{}" to Dictionary').format(word)
```

Comment:
I do not insist on keeping it “Add <word> to Dictionary” as the word is already clear from the context. Please, feel free to simplify it to “Add to Dictionary”.